tests/nosetests/pyanaconda_tests/module_subscription_tests.py

```diff
@@ -0,0 +1,261 @@
+SYSPURPOSE_VALID_VALUES_JSON_EMPTY="""
+{
+"role" : [],
+"service_level_agreement" : [],
+"usage" : []
+}
+"""
+
+class SubscriptionInterfaceTestCase(unittest.TestCase):
+    """Test DBus interface for the subscription module."""
+
+    def setUp(self):
+        """Set up the subscription module."""
+        self.subscription_module = SubscriptionModule()
+        self.subscription_interface = SubscriptionInterface(self.subscription_module)
+
+        # Connect to the properties changed signal.
+        self.callback = Mock()
+        self.subscription_interface.PropertiesChanged.connect(self.callback)
+
+    def kickstart_properties_test(self):
+        """Test kickstart properties."""
+        self.assertEqual(self.subscription_interface.KickstartCommands, ["syspurpose", "rhsm"])
+        self.assertEqual(self.subscription_interface.KickstartSections, [])
+        self.assertEqual(self.subscription_interface.KickstartAddons, [])
+        self.callback.assert_not_called()
+
+    def system_purpose_data_test(self):
+        """Test the SystemPurposeData DBus structure."""
+
+        # create the SystemPurposeData structure
+        system_purpose_data = SystemPurposeData()
+        system_purpose_data.role = "foo"
+        system_purpose_data.sla = "bar"
+        system_purpose_data.usage = "baz"
+        system_purpose_data.addons = ["a", "b", "c"]
+
+        # create its expected representation
+        expected_dict = {
+            "role" : get_variant(Str, "foo"),
+            "sla" : get_variant(Str, "bar"),
+            "usage" : get_variant(Str, "baz"),
+            "addons" : get_variant(List[Str], ["a", "b", "c"])
+        }
+
+        # compare the two
+        self.assertDictEqual(SystemPurposeData.to_structure(system_purpose_data), expected_dict)
+
+    def default_property_values_test(self):
+        """Test the default subscription module values are as expected."""
+        structure = self.subscription_interface.SystemPurposeData
+        system_purpose_data = SystemPurposeData.from_structure(structure)
+        self.assertEqual(system_purpose_data.role, "")
+        self.assertEqual(system_purpose_data.sla, "")
+        self.assertEqual(system_purpose_data.usage, "")
+        self.assertEqual(system_purpose_data.addons, [])
+        # check the lists of valid values look reasonably sane:
+        # - they are either an empty list
+        # - or they contain a non zero amount of items
+        valid_roles = self.subscription_interface.GetValidRoles()
+        valid_slas = self.subscription_interface.GetValidSLAs()
+        valid_usage_types = self.subscription_interface.GetValidUsageTypes()
+        self.assertTrue(len(valid_roles) > 0 or valid_roles == [])
+        self.assertTrue(len(valid_slas) > 0 or valid_slas == [])
+        self.assertTrue(len(valid_usage_types) > 0 or valid_usage_types == [])
+
+    def set_system_purpose_test(self):
+        """Test if setting system prupose data from DBUS works correctly."""
```

Comment:
Fixed. :)

Reply:
On a second thought, it might be a good idea to grep for that? You have the same in name of this PR, so it's probably something you (mis)type automatically without thinking...

Reply:
Yep, should be fixed now as far as I can tell. :)